src/basic/tmpfile-util.c

```diff
@@ -19,6 +19,47 @@
 #include "tmpfile-util.h"
 #include "umask-util.h"
 
+int open_temporary_child_at(int dir_fd, const char *extra, char **ret_path) {
+        _cleanup_free_ char *t = NULL;
+        _cleanup_close_ int fd = -1;
+        int r;
+
+        assert(dir_fd >= 0 || dir_fd == AT_FDCWD);
+        assert(extra);
+
+        r = tempfn_random(extra, NULL, &t);
+        if (r < 0)
+                return r;
+
+        fd = openat(dir_fd, t, O_CLOEXEC|O_NOCTTY|O_RDWR|O_CREAT|O_EXCL, 0600);
```

Comment:
O_NOCTTY together with O_CREAT|O_EXCL feels a bit unnecessary. the whole call can only work if the thing doesn't exist yet, and is guarantteed to return a regular file, hence it definitely is not a tty